diff --git a/cc/private/compile/cc_compilation_helper.bzl b/cc/private/compile/cc_compilation_helper.bzl
index 88d7404..37f6d6b 100644
--- a/cc/private/compile/cc_compilation_helper.bzl
+++ b/cc/private/compile/cc_compilation_helper.bzl
@@ -431,3 +431,5 @@ def _init_cc_compilation_context(
     else:
-        # Do not add system_include_dirs and include_dirs directly to compilation context.
+        # Do not add include dirs directly to the regular compilation context; all public search
+        # paths for external repositories are reclassified as external includes.
+        quote_include_dirs_for_context = []
         system_include_dirs_for_context = []
@@ -438,3 +440,3 @@ def _init_cc_compilation_context(
         external_include_dirs.append(bin_include_dir)
-        external_include_dirs.extend(quote_include_dirs_for_context)
+        external_include_dirs.extend(quote_include_dirs)
         external_include_dirs.extend(system_include_dirs)
