diff --git a/BUILD.bazel b/BUILD.bazel
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -133,4 +133,4 @@ ISYSTEM = [
 ]
-
-INCLUDES = ISYSTEM + [
+
+INCLUDES = [
     ".",
@@ -305,7 +305,20 @@ filegroup(
     ),
-)
-
-# --- The library ---
-
-cc_library(
+)
+
+cc_library(
+    name = "uapi_headers",
+    hdrs = glob(["src/include/uapi/**/*.h"]),
+    strip_include_prefix = "src/include/uapi",
+)
+
+cc_library(
+    name = "override_headers_library",
+    hdrs = glob(["src/include/override/**/*.h"]),
+    strip_include_prefix = "src/include/override",
+    deps = [":uapi_headers"],
+)
+
+# --- The library ---
+
+cc_library(
     name = "libudev_objects",
@@ -346,2 +359,3 @@ cc_library(
     copts = COPTS,
+    features = ["-external_include_paths"],
     includes = INCLUDES,
@@ -361,2 +375,3 @@ cc_library(
     ],
+    deps = [":override_headers_library"],
     alwayslink = True,
